client/X11/xf_video.c

```diff
@@ -652,6 +655,7 @@ void xf_video_control_init(xfContext *xfc, VideoClientContext *video)
 void xf_video_control_uninit(xfContext *xfc, VideoClientContext *video)
 {
 	xf_video_free(xfc->xfVideo);
+	xfc->xfVideo = NULL;
```

Comment:
Free here is unnecessary, post_disconnect is always called.
Or could the whole stuff be moved to the control init/uninit avoiding runtime impact when not used?